subiquity/models/filesystem.py

```diff
@@ -976,14 +1110,77 @@ def _probe_bootloader(self):
 
     def __init__(self):
         self.bootloader = self._probe_bootloader()
-        self._disk_info = []
+        self._probe_data = None
         self.reset()
 
     def reset(self):
-        self._actions = [
-            Disk.from_info(self, info) for info in self._disk_info]
+        if self._probe_data is not None:
+            config = storage_config.extract_storage_config(self._probe_data)
+            self._actions = self._actions_from_config(
+                config["storage"]["config"],
+                self._probe_data['blockdev'])
+        else:
+            self._actions = []
         self.grub_install_device = None
 
+    def _actions_from_config(self, config, blockdevs):
+        byid = {}
+        objs = []
+        exclusions = set()
+        for action in config:
+            if action['type'] == 'mount':
+                exclusions.add(byid[action['device']])
+                continue
+            c = _type_to_cls.get(action['type'], None)
+            if c is None:
+                # Ignore any action we do not know how to process yet
```

Comment:
I wonder what to do when a system has a bcache then?  The backing and caching devices will show up a disks; if they are "preserved" or just ignored (not used by the install path), then the bcache could get in the way, either during the install or after reboot.

Reply:
It's certainly not so great to show a device as "unused" when probert and curtin know it's being used in a bcache. That said, if they are used in the install, curtin will know how to dismantle the bcache. If they are not used in the install, then yes, they will still hang around, but I can't really see how that could be a problem?

I'm not sure there is really scope to do anything different here.

Reply:
I did realize thinking about this that I also need to ignore actions whose dependencies have been ignored, e.g. the filesystem of a bcache.

Reply:
Yes, I think this is one of my biggest fears;  how do we inform the users *which* devices are being destroyed/cleared.    I somewhat worry that a raid or bcache or some other multi-device element is being selected/used and then curtin will *ensure* that it can use the device, which may require dismantling an associated raid or lvm or bcache.

Reply:
Well for a compound device subiquity knows about, we make it clear in the UI that it has to be torn down before the disk can be reused, so I think that's fine. For compound devices subiquity doesn't know about (e.g. bcache and zpools but probably some other things too) I'm not sure what we can do. I guess we could add very minimal support for subiquity to display that the disk or partition is a bcache member or used by a zpool. (It would be nice if linux block devices had even _heard_ of the _concept_ of a regular interface here!)